beets/importer.py

```diff
@@ -1490,7 +1500,8 @@ def albums_in_dir(path):
         if collapse_paths:
             if (not collapse_pat and collapse_paths[0] in ancestry(root)) or \
                     (collapse_pat and
-                     collapse_pat.match(os.path.basename(root))):
+                     collapse_pat.match(os.path.basename(root))) or \
+                    is_subdir_of_any_in_list(root, collapse_paths):
```

Comment:
Not having 100% thought through this, this check seems like it might be a bit redundant with `collapse_paths[0] in ancestry(root)` above. Do we still need that check if we add this new (seemingly broader) check?